torch/csrc/jit/tracer.cpp

```diff
@@ -504,6 +504,30 @@ void addInputs(
     n->addInput(none);
   }
 }
+void addInputs(
+    Node* n,
+    const char* name,
+    const c10::optional<at::Layout>& value) {
+  if (value.has_value()) {
+    detail::genericAddInput(n, static_cast<int64_t>(*value));
+  } else {
+    Graph* g = n->owningGraph();
```

Comment:
It would be nice if you factored the 8 copy-pasta's of this three-line block into it's own function (i am one of those copy-paster's...)